### PR TITLE
feat: implement Clerk provider

### PR DESCRIPTION
**Pull request by ignaciojcano**
Implements the provider for [Clerk](Clerk.com)

Their OAuth2 usage guide is found [here](https://clerk.com/docs/advanced-usage/clerk-idp)

Created the provider using the generator and used the Auth0 one as a guide

### CODE REVIEW

File: src/Clerk/Provider.php [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            'name'     => $user['name'],
+            'email'    => $user['email'],
+            'avatar'   => $user['picture'],
+        ]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getTokenFields($code)
```
Comment (atymic) [inline src/Clerk/Provider.php:87]: Is this needed or the same as the parent method?

**Review by atymic — COMMENTED** [review]
One quick comment, otherwise looks good.

File: src/Clerk/Provider.php (hunk 1 above)
Comment (ignaciojcano) [reply src/Clerk/Provider.php]: @atymic it's been addressed

**Review by lucasmichot — APPROVED** [review]
GTM @atymic

**Comment by atymic** [comment]
Please PR the website repo @ignaciojcano

**Pull request merged**